modules/container.nix

```diff
@@ -0,0 +1,131 @@
+{
+  context,
+  system,
+  config,
+  lib,
+  ...
+}: let
+  floxpkgs = context.inputs.flox-floxpkgs;
+in {
+  options.container = with lib; {
+    name = mkOption {
+      description = mdDoc ''The name of the resulting image.'';
+      type = types.str;
+      default = "floxEnv";
+    };
+
+    tag = mkOption {
+      description = lib.mdDoc ''Tag of the generated image.'';
+      type = types.str;
+      default = "latest";
+    };
+
+    # fromImage = mkOption {
+    #   description = lib.mdDoc ''The repository tarball containing the base image. It must be a valid Docker image, such as one exported by `docker save`.'';
+    #   default = null;
+    # };
+
+    # contents = mkOption {
+    #   description = lib.mdDoc ''Top-level paths in the container. Either a single derivation, or a list of derivations.'';
+    #   default = [];
+    # };
+
+    # architecture = mkOption {
+    #   description =
+    #     lib.mdDoc ''
+    #       used to specify the image architecture, this is useful for multi-architecture builds that don't need cross compiling. If not specified it will default to `hostPlatform`.'';
+    #   default = {};
+    # };
+
+    config = mkOption {
+      description = lib.mdDoc ''
+        Run-time configuration of the container. A full list of the options
+        are available at in the
+        [Docker Image Specification v1.2.0](https://github.com/moby/moby/blob/master/image/spec/v1.2.md#image-json-field-descriptions).
+        Note that config.env is not supported (use environmentVariables instead)
+      '';
+      type = types.anything;
+      default = {};
+    };
+
+    created = mkOption {
+      description = lib.mdDoc ''Date and time the layers were created.'';
+      type = types.str;
+      default = "now";
```

Comment:
Upstream defaults to `1970-01-01T00:00:01Z` (to make containers reproducible), but to me that feels weird

Reply:
I think the 0 timestamp is better for building images internally, but if we're always doing streaming images then "now" might be the better default